#! /usr/bin/env python

# Mutual camera localization configuration file

PACKAGE='sick_pose'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name             Type      Level Description             Default Min   Max

gen.add("front_wall", double_t, 0,  "X position of the wall on right", 46, 0, 70)
gen.add("back_wall", double_t,   0,"X position of the wall on left", 0,     -50,     0)
gen.add("left_wall", double_t,   0,"Y position of the wall on front", 6,     0,     50)
gen.add("right_wall", double_t,   0,"Y position of the wall on back", -3,     -50,     0)

gen.add("front_room_delim", double_t,   0,"Y position of the wall on back", 50,     0,     50)
gen.add("left_room_delim", double_t,   0,"Y position of the wall on back", 50,     0,     50)


gen.add("cluster_distance_threshold", double_t,   0,"Threshold of the distance within the cluster", 0.6,     0,     1)
gen.add("split_and_merge_threshold", double_t,   0,"Threshold for split and merge", 0.3,     0,     1)
gen.add("edge_split_and_merge_threshold", double_t,   0,"Eliminate people around cube", 0.3,     0,     1)
gen.add("min_cluster_size", double_t,   0,"Min size of the cluster", 5,     0,     15)
gen.add("max_line_length", double_t,   0,"Max size of the cluster", 3.7,     1.5,     5)
gen.add("min_line_length", double_t,   0,"Min line lenghth", 1.5,     0,     4)


gen.add("max_translation", double_t,   0,"max_translation", 1.5,     0,     1.5)
gen.add("max_tick_ghost", int_t,   0,"max_tick_ghost", 20,     0,    100)


exit(gen.generate(PACKAGE, "sick_pose", "sickPose"))
